src/Lachain.Consensus/HoneyBadger/HoneyBadger.cs

```diff
@@ -159,13 +160,34 @@ private ConsensusMessage CreateDecryptedMessage(PartiallyDecryptedShare share)
             return message;
         }
 
-        private void HandleDecryptedMessage(TPKEPartiallyDecryptedShareMessage msg)
+        // We need to handle this message carefully like how about decoding a random message with random length
+        // and the value of 'share.ShareId' needs to be checked. If it is out of range, it can throw exception
+        private void HandleDecryptedMessage(TPKEPartiallyDecryptedShareMessage msg, int senderId)
         {
-            var share = Wallet.TpkePublicKey.Decode(msg);
-            _decryptedShares[share.ShareId].Add(share);
-            CheckDecryptedShares(share.ShareId);
+            PartiallyDecryptedShare? share = null;
+            try
+            {
+                // Converting any random bytes to G1 is not possible
+                share = Wallet.TpkePublicKey.Decode(msg);
+                _decryptedShares[share.ShareId].Add(share);
+            }
+            catch (Exception exception)
+            {
+                var pubKey = Broadcaster.GetPublicKeyById(senderId)!.ToHex();
+                Logger.LogWarning($"Exception occured handling Decrypted message: {msg} from {senderId} ({pubKey})");
+            }
```

Comment:
Should we re-throw this exception to the higher level?

Reply:
Like in RootProtocol? By the way if we throw exception then HoneyBadger will stop, right? Should we do that?

Reply:
yes,  you are right, we should not stop the protocol in this case,  so we should not re-throw the error